### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by Soldem**
add task 4

### CODE REVIEW

**Comment by Soldem** [comment]
https://contest.yandex.ru/contest/47445/run-report/85357881/

File: BinarySearch/BinarySearch.sln [hunk 1]
```diff
@@ -0,0 +1,31 @@
+﻿
```
Comment (TieBeTie) [inline BinarySearch/BinarySearch.sln:1]: ?????????????

File: BinarySearch/BinarySearch/BinarySearch.cpp [hunk 2]
```diff
@@ -0,0 +1,40 @@
+﻿#include <iostream>
+
+int* MAS;
+int SIZE;
```
Comment (TieBeTie) [inline BinarySearch/BinarySearch/BinarySearch.cpp:3]: Do not use global variables

File: BinarySearch/BinarySearch/BinarySearch.cpp [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    if (MAS[left] == target) return true;
+    else return false;
+}
+
+int main()
+{
+    int q, i, j, t;
+    std::cin >> SIZE;
+
+    MAS = new int[SIZE];
```
Comment (TieBeTie) [inline BinarySearch/BinarySearch/BinarySearch.cpp:21]: memory leak(

**Pull request merged**